### PR TITLE
Add version meta data to RC templates

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Add Version Metadata to Remote Config templates
- Exclude output only parameters from metadata in input templates
- Add unit tests

### CODE REVIEW

File: test/unit/remote-config/remote-config.spec.ts [hunk 1]
```diff
@@ -596,6 +614,18 @@ describe('RemoteConfig', () => {
           }).to.throw(
             'Cannot set property etag of #<RemoteConfigTemplateImpl> which has only a getter');
 
+          const v = template.version;
+          expect(v).to.be.not.undefined;
+          const version = v as Version;
```
Comment (hiranya911) [inline test/unit/remote-config/remote-config.spec.ts:619]: Why is this needed? Shouldn't this already be typed `Version`?

File: test/unit/remote-config/remote-config.spec.ts [hunk 2]
```diff
@@ -596,6 +614,18 @@ describe('RemoteConfig', () => {
...
+          expect(version.versionNumber).to.equal('86');
```
Comment (hiranya911) [inline test/unit/remote-config/remote-config.spec.ts:620]: Can we do this:

```
expect(version).to.deep.equal(VERSION_INFO);
```

File: test/unit/remote-config/remote-config.spec.ts [hunk 3]
```diff
@@ -612,7 +642,11 @@ describe('RemoteConfig', () => {
           expect(cond.tagColor).to.equal(TagColor.BLUE);
 
           const parsed = JSON.parse(JSON.stringify(template));
-          expect(parsed).deep.equals(REMOTE_CONFIG_RESPONSE);
+          const templateCopy = deepCopy(REMOTE_CONFIG_RESPONSE);
```
Comment (hiranya911) [inline test/unit/remote-config/remote-config.spec.ts:645]: I had a little trouble following the logic here. May be rename the variables to `expectedTemplate` and `expectedVersion`.

File: test/unit/remote-config/remote-config.spec.ts (hunk 1 above)
Comment (lahirumaramba) [reply test/unit/remote-config/remote-config.spec.ts:619]: If I do not cast it here the compiler complains that the object might be undefined. I think it is because of version is optional. We do the same for conditions below...
`const cond = c as RemoteConfigCondition;` any thoughts?

File: test/unit/remote-config/remote-config.spec.ts (hunk 2 above)
Comment (lahirumaramba) [reply test/unit/remote-config/remote-config.spec.ts:620]: `version` is a `Version` type and `VERSION_INFO` is a js object so deep equal didn't work unfortunately...

File: test/unit/remote-config/remote-config.spec.ts (hunk 3 above)
Comment (lahirumaramba) [reply test/unit/remote-config/remote-config.spec.ts:645]: Ah good call! Thanks!

File: test/unit/remote-config/remote-config.spec.ts (hunk 2 above)
Comment (lahirumaramba) [reply test/unit/remote-config/remote-config.spec.ts:620]: An alternative is to define a global `Version` const an use that here

File: test/unit/remote-config/remote-config.spec.ts (hunk 1 above)
Comment (hiranya911) [reply test/unit/remote-config/remote-config.spec.ts:619]: Add an assertion.

```
const version = template.version!;
```

File: test/unit/remote-config/remote-config.spec.ts (hunk 2 above)
Comment (hiranya911) [reply test/unit/remote-config/remote-config.spec.ts:620]: If that's feasible let's do that. Otherwise the way it is now is not terrible either.

Comment (lahirumaramba) [reply test/unit/remote-config/remote-config.spec.ts:618]: Looks like it wouldn't work unless we use an instance of the internal `VersionImpl`, which I do not think we can use here.
